File: app/src/main/java/com/yalantis/data/source/repository/RepositoryRemoteDataSource.java
package com.yalantis.data.source.repository;

import android.support.annotation.NonNull;

import com.yalantis.data.Repository;
import com.yalantis.data.source.base.BaseRemoteDataSource;

import java.util.List;

import rx.Single;

/**
 * Created by irinagalata on 12/1/16.
 */

class RepositoryRemoteDataSource extends BaseRemoteDataSource implements RepositoryDataSource {

    @Override
    public Single<List<Repository>> getRepositories(@NonNull String organization) {
        return mGithubService.getOrganizationRepos(organization);
    }

    @Override
    public void saveRepositories(List<Repository> repositories) {

    }

    @Override
    public void clearRepositories() {

    }

    @Override
    public boolean isEmpty() {
        return false;
    }

}

File: app/src/main/java/com/yalantis/data/source/repository/RepositoryRemoteDataSource.kt
package com.yalantis.data.source.repository

import com.yalantis.data.Repository
import com.yalantis.data.source.base.BaseRemoteDataSource
import io.reactivex.Single

/**
 * Created by irinagalata on 12/1/16.
 */

internal class RepositoryRemoteDataSource : BaseRemoteDataSource(), RepositoryDataSource {

    override fun getRepositories(organization: String): Single<List<Repository>> =
            githubService.getOrganizationRepos(organization)

    override fun saveRepositories(repositories: List<Repository>) {

    }

    override fun clearRepositories() {

    }

    override fun isEmpty(): Boolean = false
}
